admin/listPurchases/listView.html

```diff
@@ -123,33 +114,30 @@
     return currentPage - centerOffset;
 }
 
-function renderSelectors(needAdvancedNav, startIndex) {
-    for(let i = startIndex; (i < startIndex + amountOfShownPages) && (i < amountOfPages); i++) {
-        navHolder.innerHTML += `<a href="#" class="option page-index-${i}-selector">${i + 1}</a>`;
+function renderSubpageSelectors() {
+    const selectors = [];
+    const startIndexToRender = calculateStartIndex();
+
+    for(let i = startIndexToRender; (i < startIndexToRender + amountOfShownPages) && (i < amountOfPages); i++) {
```

Comment:
You could calculate `const endOfPages = Math.min(startIndexToRender + amountOfShownPages, amountOfPages)` to allow a simpler check in the for-loop: `i < endOfPages`.